modules/islandora_scholar_embargo/includes/admin.form.inc

```diff
@@ -21,7 +21,9 @@ function islandora_embargo_admin(array $form, array &$form_state) {
   $options = islandora_get_content_models();
   $selected = variable_get('islandora_embargo_content_models', array('ir:citationCModel', 'ir:thesisCModel'));
   foreach ($selected as $cmodel) {
-    $options = array($cmodel => $options[$cmodel]) + $options;
+    if (isset($options[$cmodel])) {
+      $options = array($cmodel => $options[$cmodel]) + $options;
+    }
```

Comment:
I'm... not sure I understand what's going on here. If there's an entry in `$options`, we reset its value to itself?

Are we looking for something more like:
```php
$options = array_intersect_key($options, drupal_map_assoc($selected));
```
instead of this `foreach` over `$selected`?

... Actually... This was some craziness to establish the "desired" ordering or something, was it?